add ECS support (#137)
logstash-plugins/logstash-input-http: docs/index.asciidoc

and applications. By taking advantage of the vast plugin ecosystem available in Logstash
you can trigger actionable events right from your application.

[id="plugins-{type}s-{plugin}-ecs_metadata"]
==== Event Metadata and the Elastic Common Schema (ECS)

> In addition to decoding the events, this input will add HTTP headers of the connection to each event.

Reviewer: ```suggestion
In addition to decoding the events, this input will add HTTP headers containing connection information to each event.

```

> When ECS compatibility is disabled, the headers are stored in the `headers` field, which had the potential to create confusion and schema conflicts downstream.

Reviewer: ```suggestion
When ECS compatibility is disabled, the headers are stored in the `headers` field, which has the potential to create confusion and schema conflicts downstream.
```

> When ECS is enabled, we can ensure a pipeline still has access to this metadata throughout the event's lifecycle without polluting the top-level namespace.

Reviewer: ```suggestion
When ECS is enabled, we can ensure a pipeline maintains access to this metadata throughout the event's lifecycle without polluting the top-level namespace.
```

The list of ciphers suite to use, listed by priorities.

[id="plugins-{type}s-{plugin}-ecs_compatibility"]
===== `ecs_compatibility`

* Value type is <<string,string>>
* Supported values are:
** `disabled`: unstructured connection metadata added at root level
** `v1`: headers added under `[@metadata][http][header]`. Some are copied to structured ECS fields `http`, `url`, `user_agent` and `host`

Controls this plugin's compatibility with the
{ecs-ref}[Elastic Common Schema (ECS)].
See <<plugins-{type}s-{plugin}-ecs_metadata>> for detailed information.

> Example output:
> 
> ECS disabled

Reviewer: ```suggestion
**Sample output: ECS disabled**
```

[source,text]
-----
{
    "@version" => "1",
    "headers" => {
           "request_path" => "/twitter/tweet/1",
            "http_accept" => "*/*",
           "http_version" => "HTTP/1.1",
         "request_method" => "PUT",
              "http_host" => "localhost:8080",
        "http_user_agent" => "curl/7.64.1",
         "content_length" => "5",
           "content_type" => "application/x-www-form-urlencoded"
    },
    "@timestamp" => 2021-05-28T19:27:28.609Z,
    "host" => "127.0.0.1",
    "message" => "hello"
}
-----

> ECS enabled

Reviewer: ```suggestion
**Sample output: ECS enabled**
```

Reviewer: Thanks for these wonderful examples.  I suggested a formatting change to give them more emphasis.
